Improve woo email content placeholder block

triple0t opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes #56016  Closes WOOPLUG-3313
This PR improves the WooCommerce Email Editor functionality by enhancing the placeholder content block integration and adding new customization options for the email editor sidebar.

#### Key Changes
1. **Email Editor Placeholder Block**
   - Added new functionality for rendering dummy order content in the email preview
   - Implemented hover state and visual feedback for the placeholder block
   - Created a dedicated preview system for different email types

2. **Sidebar Improvements**
   - Enhanced Email Type Info Component with customizable icon and content
   - Added new filters for customizing sidebar information:
     - `woocommerce_email_editor_sidebar_email_type_info_icon`
     - `woocommerce_email_editor_sidebar_email_type_info_content`

3. **Preview and Configuration Updates**
   - Updated email preview links and configuration
   - Added support for dynamic email type previews
   - Improved preview rendering with proper header/footer handling

4. **Documentation**
   - Updated README with new filter documentation
   - Added changelog entries for the improvements


(For Bug Fixes) Bug introduced in PR # .

### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

| Before | After |
| ------ | ----- |
|   ![Screenshot 2025-03-10 at 12 22 05 PM](https://github.com/user-attachments/assets/bfb8dfbb-36c7-4448-8d9d-cf1b6c22f99a)     |   <img width="1195" alt="Screenshot 2025-03-13 at 1 07 24 PM" src="https://github.com/user-attachments/assets/21d89d01-adb8-4fe0-a77d-237489ecfae7" /> |


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the `block_email_editor` experimental flag (in DB, set `woocommerce_feature_block_email_editor_enabled` option to `yes` or WP CLI `wp option add woocommerce_feature_block_email_editor_enabled yes`).
2. Navigate to `WP Admin > Woo Emails`
3. Create or edit an email template
4. Verify the placeholder block shows proper preview content
5. Check hover states and interactions
6. Test different email types and preview functionality
7. Verify sidebar customizations work as expected

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

triple0t committed:
Update email editor wooContentBlock placeholder

* Add 'id' property to email types in global.d.ts for better identification.
* Introduce block_preview_url in global.d.ts to facilitate email preview functionality.
* Refactor woo_content_placeholder_block.tsx to use WooContentPlaceholderEditContent for editing email content.
* Create new component WooContentPlaceholderEditContent.tsx to act as email content placeholder preview.

This update improves the structure and functionality of the email editor integration, allowing for a more dynamic editing experience.

triple0t committed:
Enhance Email Type Info Component in Sidebar

* Introduce filters for customizing the email type info icon and content.
* Update the content description to clarify the purpose of the email block.

These changes improve the user experience by providing clearer visual and textual information in the email editor's sidebar.

triple0t committed:
Update email preview links and add configuration filter

* Change the sending method configuration link to point to MailPoet's SMTP plugin documentation.
* Update the sign-up link for MailPoet to reflect the WooCommerce email editor context.
* Introduce a new filter for the sending method configuration link in the WooCommerce email editor.

These changes enhance the user experience by providing more relevant resources and improving link accuracy in the email editor.

triple0t committed:
Add preview email editor placeholder block dummy content functionality

* Introduce a new method to handle previewing email editor placeholder block content with security checks.
* Implement a capture function to render WooCommerce email content without headers and footers.
* Enhance the email preview experience by allowing users to view email types dynamically.

These changes improve the email editor's functionality, providing a better user experience for previewing email content.

triple0t committed:
Refactor WooCommerce email editor placeholder content block integration

* Replace the existing wooContentBlock with a new structure using wooContentPlaceholderBlock for better organization.
* Introduce new components for editing and saving email content, enhancing the email editor's functionality.
* Remove outdated files related to the previous block structure, streamlining the codebase.

These changes improve the maintainability and usability of the email editor, providing a more dynamic editing experience for users.

triple0t committed:
Add changelog files

triple0t committed:
Fix php lint errors

* Improve documentation for the email preview instance and capture function.
* Refactor code for better readability and maintainability in the email editor.

lysyjan commented on plugins/woocommerce/client/admin/client/wp-admin-scripts/email-editor-integration/blocks/woo-email-content/placeholder-block-edit-content.tsx:
```diff
@@ -0,0 +1,127 @@
+				style={ {
+					width: '100%',
+					height:
+						iframeRef?.current?.contentWindow?.outerHeight ||
```
I am not sure about the iframe height.
When I tried to use, for example, a content for the reset password email. It does not look good. But maybe we are okay with that because it is only a temporary solution. What do you think?
<img width="678" alt="Screenshot 2025-03-17 at 15 21 00" src="https://github.com/user-attachments/assets/832c634a-69af-42f9-98b5-3c368ee65030" />

triple0t replied:
Fixed in https://github.com/woocommerce/woocommerce/pull/56365/commits/56159aba4957726820dbf71ae702d39bea047ec3

lysyjan reviewed (commented):
Great job @triple0t! 

I have a few questions about UI etc.
1. The iframe contains a default Woo content but the select does not contain selected type. It seems to be a bit confusing to me. Maybe the value from select should also contain preselected value.
2. In comparison with the preview in Woo -> Settings -> Emails the heading for all emails is missing. Can we add it? For example, when the heading `Thank you for your order`. Maybe it is okay. I am asking because it is one of the main differences.
3. This paragraph seems a bit redundant now. As a user, I need to remove or edit it.
4. The email text is aligned to the middle. Again, maybe this is fine, but it is one of the main differences between the preview in Woo -> Settings -> Emails.
<img width="1122" alt="Screenshot 2025-03-17 at 15 27 47" src="https://github.com/user-attachments/assets/c777fdff-b23b-4682-9f54-53e3262da705" />

triple0t committed:
Refactor WooCommerce email editor placeholder block styles and functionality

* Replace Surface component with a styled div for improved visual consistency. This removes the unnecessary dependency on an experimental component
* Update iframe background color handling to ensure proper display of email content.
* Enhance the dynamic height adjustment of the iframe based on the email content.

These changes enhance the user experience by providing a more polished and responsive email editing interface.

triple0t commented:
Thanks for the review, @lysyjan 

> The iframe contains a default Woo content but the select does not contain selected type. It seems to be a bit confusing to me. Maybe the value from select should also contain preselected value.

Unfortunately, we can't preselect the option as the Editor autosave mechanism will save the type association. As this will update the transaction email template for the users' store, we only want to do this when initiated by the user.

We will also be updating the select control in https://github.com/woocommerce/woocommerce/issues/56109, so it's best to leave it as it is for now.

> In comparison with the preview in Woo -> Settings -> Emails the heading for all emails is missing. Can we add it? For example, when the heading Thank you for your order. Maybe it is okay. I am asking because it is one of the main differences.

Yes, it was intentionally removed. We will be updating the block more in https://github.com/woocommerce/woocommerce/issues/56109


> This paragraph seems a bit redundant now. As a user, I need to remove or edit it.
The email text is aligned to the middle. Again, maybe this is fine, but it is one of the main differences between the preview in Woo -> Settings -> Emails.

We plan to remove these blocks of text from the placeholder content (and generated emails) in https://github.com/woocommerce/woocommerce/issues/56109

General idea: The user-editable part of the email template will be moved (done) in the email editor. The dynamic part will remain the same.

lysyjan reviewed (approved):
I tested the new functionality, and it works well.

<img width="1569" alt="Screenshot 2025-03-19 at 9 04 24" src="https://github.com/user-attachments/assets/ef372145-2cb8-4041-8de4-1707dc6f44fc" />
